calyx-py/calyx/builder.py

```diff
@@ -706,6 +706,20 @@ def sub_store_in_reg(self, left, right, cellname, width, ans_reg=None):
         return sub_group, ans_reg
 
 
+class CellAndGroup:
```

Comment:
This is a perfect use case for [dataclasses](https://docs.python.org/3/library/dataclasses.html), preferably with `frozen=True` because we never expect to mutate this stuff.